### ISSUE TITLE
SvelteToast is not a valid SSR component

### ISSUE THREAD

**Issue by Iuriy-Budnikov** [zerodevx/svelte-toast#60]
I'm getting this error with sveltekit ssr

`<SvelteToast>` is not a valid SSR component. You may need to review your build config to ensure that dependencies are compiled, rather than imported as pre-compiled modules.

**Comment by zerodevx** [comment]
This component historically works with SSR. Could you help check by installing the previous minor:

```
$ npm i --save-exact @zeroodevx/svelte-toast@0.7.2
```

and see if the error still happens?

**Comment by Iuriy-Budnikov** [comment]
> This component historically works with SSR. Could you help check by installing the previous minor:
> 
> ```
> $ npm i --save-exact @zeroodevx/svelte-toast@0.7.2
> ```
> 
> and see if the error still happens?

Yeah, I'm getting the same error.

**Comment by zerodevx** [comment]
It's likely due to the [deprecation of `svelte` field](https://github.com/sveltejs/kit/pull/7489) that landed in SvelteKit a few days ago. Looks like I'll have to split the package into two - one for Svelte, and one for vanilla JS.

**Comment by jrmoynihan** [comment]
> It's likely due to the [deprecation of `svelte` field](https://github.com/sveltejs/kit/pull/7489) that landed in SvelteKit a few days ago. Looks like I'll have to split the package into two - one for Svelte, and one for vanilla JS.

That is indeed part of the reason.  I just started patching this in some other packages/forks I use.  You can save a lot of headaches by using SvelteKit itself to create the components and then just running the built-in [`svelte-package`](https://kit.svelte.dev/docs/packaging) command to do all the work for you when it comes to creating the correct package.json, exporting types, and ensuring SSR compatibility.  It makes staying up-to-date with the changing API a lot more seamless.

I'm not sure how that would play for the vanilla JS consumers of this package, but I don't frequently see a lot of comments/issues from those users, so that's an encouraging sign.

**Comment by zerodevx** [comment]
@Grandnainconnu thanks for the gist. 🙏 

@jrmoynihan thanks for the recommendation. Migrating to `svelte-package` is indeed the immediate next step. Will take the opportunity to release the next `major` as well; deprecate outdated stuff, refactor some code, migrate e2e tests to `playwright`, implement enhancements logged in issues, etc - hopefully I can start this sooner rather than later.

Because of this change, `svelte` consumers of this package now picks up the compiled code rather than the svelte definitions and it's incredibly inefficient.

> I'm not sure how that would play for the vanilla JS consumers of this package, but I don't frequently see a lot of comments/issues from those users, so that's an encouraging sign.

Personally, I use the vanilla compiled code for some projects - but it's not complicated to publish a vanilla build as well.

**Comment by Iuriy-Budnikov** [comment]
@zerodevx 

Now I'm getting an error 

```
Cannot find module 'node_modules/@zerodevx/svelte-toast/stores' imported from node_modules/@zerodevx/svelte-toast/index.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'node_modules/@zerodevx/svelte-toast/stores' imported from node_modules/@zerodevx/svelte-toast/index.js
```

**Comment by zerodevx** [comment]
How did you get the error? I just loaded `v1.0.0-rc.1` in a new SvelteKit clean install and works.

https://stackblitz.com/edit/sveltejs-kit-template-default-7kksrr?file=src/routes/+layout.svelte

**Comment by zerodevx** [comment]
Fixed by #66. Just upgrade to latest stable `v0.9`.
